blog/_posts/2024-08-29-condaEnvs.md

```diff
@@ -0,0 +1,164 @@
+---
+layout: page
+title:  Recording and exporting conda environments for Python
+date:   2024-08-29
+subheadline: "Training"
+teaser: "
+ Knowing how best to balance reproducibility and reuseability can be difficult when exporting conda environments. How can you ensure dependencies are recorded but also ensure restrictive versioning doesn't prevent you from rebuilding a conda env?
+"
+author: Maeve Murphy Quinlan
+image:
+  thumb: research_hack.jpg
+  caption: Managing conda environments can be the difference between smooth sailing and dependency hell
+---
+
+# Recording and exporting conda environments for Python
+
+When it comes to version and package management for Python, there are an overwhelming number of options, including minimal approaches including [pyenv](https://github.com/pyenv/pyenv), [pipenv](https://pypi.org/project/pipenv/), and [venv](https://docs.python.org/3/library/venv.html); conda-centric approaches including [Anaconda](https://www.anaconda.com/), [Miniconda](https://docs.anaconda.com/miniconda/), and [Miniforge](https://github.com/conda-forge/miniforge), and multi-featured package and project-management tools such as [Pixi](https://pixi.sh/latest/) and [Poetry](https://python-poetry.org/).
+
+Depending on your specific use-case, some of these tools may be better suited to your project; for example, if you are developing a package and want to build your binaries locally, Poetry might be a great solution for you. If you want to install scientific packages from certain conda channels, one of the conda ecosystem options might suit you best.
+
+Let's assume you are working within a research group that generally uses conda envs, and you are installing software that frequently includes conda-related install instructions. We recommend using [Miniforge](https://github.com/conda-forge/miniforge), the open source Mamba-enabled tool that is essentially drop-and-replace with conda.
```

Comment:
I don't think this is right.  Mambaforge was the drop in replacement that's mamba enabled.  Miniforge is just the version of miniconda that only defaults to using conda-forge, so avoids license issues of the anaconda repositories.

Reply:
So I thought this, but then I got a bit confused by the release notes that imply mambaforge and miniforge are now basically the same?

https://github.com/conda-forge/miniforge

> After the release of Miniforge 23.3.1 in August 2023, Miniforge and Mambaforge are essentially identical. The only difference is the name of the installer and subsequently the default installation path.

> Before that release, Miniforge only shipped conda, while Mambaforge added mamba on top. Since Miniconda started shipping conda-libmamba-solver in July 2023, Miniforge followed suit and started shipping it too in August. At that point, since conda-libmamba-solver depends on libmambapy, the only difference between Miniforge and Mambaforge was the presence of the mamba Python package. To minimize surprises, we decided to add mamba to Miniforge too.

I wasn't sure how best to phrase the above. Will add in a comment on the licensing too, and maybe find a good link to somewhere discussing it.

Reply:
No that's fine, I'm out of date.